src/bidiMapper/modules/network/NetworkRequest.ts

```diff
@@ -304,26 +304,66 @@ export class NetworkRequest {
   }
 
   get #timings(): Network.FetchTimingInfo {
+    // The timing in the CDP events are provided relative to the event's baseline.
+    // However, the baseline can be different for different events, and the events have to
+    // be normalized throughout resource events. Normalize events timestamps  by the
+    // request.
+    // TODO: Verify this is correct.
+    const responseTimeOffset = getTiming(
+      getTiming(this.#response.info?.timing?.requestTime) -
+        getTiming(this.#request.info?.timestamp),
+    );
+
     return {
       // TODO: Verify this is correct
-      timeOrigin: getTiming(this.#response.info?.timing?.requestTime),
-      requestTime: getTiming(this.#response.info?.timing?.requestTime),
+      timeOrigin: Math.round(getTiming(this.#request.info?.wallTime) * 1000),
+      // Timing baseline.
+      // TODO: Verify this is correct.
+      requestTime: 0,
+      // TODO: set if redirect detected.
       redirectStart: 0,
+      // TODO: set if redirect detected.
       redirectEnd: 0,
       // TODO: Verify this is correct
       // https://source.chromium.org/chromium/chromium/src/+/main:net/base/load_timing_info.h;l=145
-      fetchStart: getTiming(this.#response.info?.timing?.requestTime),
-      dnsStart: getTiming(this.#response.info?.timing?.dnsStart),
-      dnsEnd: getTiming(this.#response.info?.timing?.dnsEnd),
-      connectStart: getTiming(this.#response.info?.timing?.connectStart),
-      connectEnd: getTiming(this.#response.info?.timing?.connectEnd),
-      tlsStart: getTiming(this.#response.info?.timing?.sslStart),
-      requestStart: getTiming(this.#response.info?.timing?.sendStart),
+      fetchStart: getTiming(
```

Comment:
I think these have to be durations from timeOrigin

Reply:
I assume CDP `response.timing` is relative to the baseline CDP `response.timing.requestTime`. The `responseTimeOffset` represents the diff between request and response baseline. It's my speculations which ends up to a meaningful result.